scripts/vaccination_alerts/send-emails.ts

```diff
@@ -0,0 +1,196 @@
+    console.log(`Emails sent: ${emailSent}`);
+    console.log(`Invalid emails removed: ${invalidEmailCount}`);
+  }
+
+  async updateVaccinationInfoVersion(alert: RegionVaccinePhaseInfo) {
+    return this.firestoreSubscriptions.updateVaccinationInfoVersion(
+      alert.fips,
+      alert.emailAlertVersion,
+    );
+  }
+
+  async sendAlertEmail(
+    email: string,
+    alert: RegionVaccinePhaseInfo,
+    dryRun: boolean = false,
+  ): Promise<void> {
+    if (dryRun) {
+      this.updateSentEmailCounters(email);
+      return;
+    }
+
+    let sendData: EmailSendData = generateEmailData(email, alert.fips);
+
+    try {
+      await this.emailService.sendEmail(sendData);
+      await this.onEmailSent(email, alert);
+    } catch (err) {
+      if (isInvalidEmailError(err)) {
+        await this.onInvalidEmail(email);
+      } else {
+        const { fips, emailAlertVersion } = alert;
+        console.error(
+          `Error sending email to ${email}, FIPS: ${fips}, version: ${emailAlertVersion}.`,
+          err,
+        );
+        this.errorCount += 1;
+      }
+    }
+  }
+}
+
+/**
+ * Send vaccination alerts to subscribers. Fetch the list of users to email from
+ * Firestore and send the emails in batches. It takes optional parameters:
+ *
+ * - `dryRun`: if true, it doesn't send the emails, just logs the number of
+ *   emails to be sent.
+ * - `emailAddress`: if given, it only send the email to that address, instead of
+ *   sending the emails to the subscribers in Firestore.
+ *
+ * yarn vaccinations-send-alert-emails [dryRun] [emailAddress]
+ *
+ */
+async function main(alertPath: string, dryRun: boolean, singleEmail?: string) {
+  const alertsByFips = readVaccinationAlerts(alertPath);
+  const alerts = _.values(alertsByFips);
+
+  const alertEmailService = new SendVaccinationAlertsService();
+
+  const emailAlertTuples = _.flatten(
+    await Promise.all(
+      alerts.map(async alert => {
+        const emails = await alertEmailService.fetchEmailsForAlert(alert);
+        return (
+          emails
+            // if an email is specified, filter out all other emails. Used for testing
+            .filter(email => (singleEmail ? email === singleEmail : true))
+            .map(email => ({ email, alert }))
+        );
+      }),
+    ),
+  );
+
+  const emailBatches = _.chunk(emailAlertTuples, BATCH_SIZE);
+  console.log(
+    `Sending ${emailAlertTuples.length} emails in ${emailBatches.length} batches.`,
+  );
+
+  for (const batch of emailBatches) {
+    await Promise.all(
+      batch.map((data: { email: string; alert: RegionVaccinePhaseInfo }) =>
+        alertEmailService.sendAlertEmail(data.email, data.alert, dryRun),
+      ),
+    );
+  }
+
+  alertEmailService.logEmailsToBeSentStats();
+
+  if (dryRun) {
+    return;
+  }
+
+  // Exit with error if there is any errors sending the emails
+  if (alertEmailService.errorCount >= 1) {
+    alertEmailService.logErrors();
```

Comment:
Hrm.  As written, this is only going to log the "Invalid emails removed" message if there were _other_ errors as well.  I think maybe we should just call logErrors() always and have it log both the fatal and non-fatal errors?  Not sure.